src/invidious/helpers/youtube_api.cr

```diff
@@ -98,13 +177,33 @@ end
 # The requested data is an endpoint (URL without the domain part)
 # and the data as a Hash object.
 #
-def _youtube_api_post_json(endpoint, data)
+def _youtube_api_post_json(endpoint, data, client_type = ApiClients::Web,
+                           proxy_region : String | Nil = nil) : Hash(String, JSON::Any)
+  client_key = HARDCODED_CLIENTS[client_type]["key"]
+  default_client_key = HARDCODED_CLIENTS[ApiClients::Web]["key"]
+
   # Send the POST request and parse result
-  response = YT_POOL.client &.post(
-    "#{endpoint}?key=#{HARDCODED_API_KEY}",
-    headers: HTTP::Headers{"content-type" => "application/json; charset=UTF-8"},
-    body: data.to_json
-  )
+  if proxy_region
+    response = YT_POOL.client(proxy_region, &.post(
+      "#{endpoint}?key=#{client_key || default_client_key}",
+      headers: HTTP::Headers{
+        "content-type"  => "application/json; charset=UTF-8",
+        "Accept-Encoding": "gzip",
+      },
```

Comment:
as those are redundant across the two if/else branches, you should declare a `url` and a `headers` variable before the if/else, for code clarity/simplicity.